STUDIO/org.openl.rules.webstudio/src/org/openl/rules/webstudio/web/repository/DeploymentController.java

```diff
@@ -393,6 +395,26 @@ public String openSelectedProjects() {
         return null;
     }
 
+    private RulesProject getRulesProject(UserWorkspace workspace,
+            DeploymentDescriptorItem deployment) throws ProjectException {
+        try {
+            return workspace.getProject(deployment.getRepositoryId(), deployment.getName(), false);
+        } catch (ProjectException e) {
+            Repository repo = workspace.getDesignTimeRepository().getRepository(deployment.getRepositoryId());

```

Comment:
deployment.getRepositoryId() can be null if Deploy Configuration was created in previous version of WebStudio (and in that version repositoryId was absent).
If repository id is null we have 2 options: 1) we can get first repository from repositories list or 2) we just rethrow e, it's more meaningful than NPE

Reply:
fixed